hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -405,6 +405,12 @@ private long handleDirectoryCleanUp(
         while (deletedDirIterator.hasNext()) {
           Table.KeyValue<String, OmKeyInfo> deletedDir =
               deletedDirIterator.next();
+          String deletedDirKey = deletedDir.getKey();
+
+          // Exit for dirs out of snapshot scope.
+          if (!deletedDirKey.startsWith(dbBucketKeyForDir)) {
```

Comment:
Should we use [Table#iterator(prefix)](https://github.com/apache/ozone/blob/4b9b10f6206ed5c29c0c4957a651de1d24fe86f6/hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/Table.java#L174)?

Reply:
I was not aware this existed. Yes, we can do it, but we need to change this in multiple places and test it. I can do it as a part of a larger separate patch.